### PR TITLE
Updates for clang 7.0

### PR DESCRIPTION
**Pull request by mingwandroid**
Not sure you want to update yet, but if so this PR may be of use.

### CODE REVIEW

**Comment by mingwandroid** [comment]
Clang 7 is not API compatible with 6 here unfortunately. I decided against adding ugly logic based on some define or other to support both. If you prefer that let me know and I'll rework this.

File: clang_delta/ExpressionDetector.cpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ class IncludesPPCallbacks : public PPCallbacks {
                           StringRef FileName, bool IsAngled,
                           CharSourceRange FilenameRange, const FileEntry *File,
                           StringRef SearchPath, StringRef RelativePath,
-                          const Module *Imported) override;
+                          const Module *Imported);
```
Comment (chenyang78) [inline clang_delta/ExpressionDetector.cpp]: This doesn't look correct. We should follow the same API as in LLVM 7.0, e.g.,

    https://github.com/csmith-project/creduce/commit/c0d9c3a2ef763eb09f0a70b8ad86250598549487

**Comment by chenyang78** [comment]
@mingwandroid Thanks for the patch. The rest of the patch looks good to me. @eeide may need to take a look at it, too.

File: clang_delta/ExpressionDetector.cpp (hunk 1 above)
Comment (mingwandroid) [reply clang_delta/ExpressionDetector.cpp]: Well, if you believe clang is wrong to give that error in this case then we need to disable that error for clang in the build scripts.

Comment (mingwandroid) [reply clang_delta/ExpressionDetector.cpp]: Oh I see, yes.

**Comment by regehr** [comment]
thanks!

**Comment by regehr** [comment]
closing this PR and merging in a different one

**Pull request closed without merging**